### DISCUSSION TITLE
GitHub App permissions for GraphQL query

### DISCUSSION THREAD

**Discussion by shipit** [community/community#24797] [Apps, API and Webhooks]
<p>I had this query working on my OAuth app with <code>user:email, repo</code> scope. It is returning <code>403 Forbidden</code> when I auth with GitHub App. Query-</p>
<pre><code>{
  viewer {
    repositories(first: 100, ownerAffiliations: OWNER) {
      totalCount
      nodes {
        nameWithOwner
        owner {
          avatarUrl
          login
        }
        languages(first: 10) {
          nodes {
            name
          }
        }
        defaultBranchRef {
          name
          target {
            ... on Commit {
              committedDate
              oid
              message
            }
          }
        }
      }
    }
  }
}
</code></pre>
<p>Permissions on the GitHub app-</p>
<pre><code>"permissions":{"contents":"read","metadata":"read","pull_requests":"read","repository_hooks":"write","statuses":"read"}
</code></pre>
<p>Thanks!</p>

**Comment by shipit** [comment]
<p>From here looks like <code>metadata</code> should be enough-</p>
<p><a href="https://developer.github.com/v3/apps/permissions/" rel="nofollow noopener">https://developer.github.com/v3/apps/permissions/</a></p>

**Comment by shipit** [comment, marked as answer]
<p>The reason this query doesn’t work is that the /repos endpoint returns public repos for the GitHub App’s token and the nested query eg /languages breaks the graphql query. The solution is to fetch <code>/user/:install_id/repos/</code> and then iterate over returned queries. </p>
<p>imho- public repos shouldn’t be returned for GitHub App in the first place as the app could be installed on a public repo in future.</p>

**Comment by PabloSzx** [comment]
<p>How can I do the query you mention in  Github API v4?</p>
